sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/conditionalExpressions.scala

```diff
@@ -294,7 +294,8 @@ object CaseWhen {
       case cond :: value :: Nil => Some((cond, value))
       case value :: Nil => None
     }.toArray.toSeq  // force materialization to make the seq serializable
-    val elseValue = if (branches.size % 2 == 1) Some(branches.last) else None
+    // avoid false positive of IM_BAD_CHECK_FOR_ODD by SpotBugs
```

Comment:
I feel that this comment might be a bit confusing to readers. Is it good without this comment?